src/ngraph/graph_util.cpp

```diff
@@ -399,6 +399,11 @@ void ngraph::insert_new_node_between(const shared_ptr<Node>& src_node,
                                      const shared_ptr<Node>& dst_node,
                                      const shared_ptr<Node>& new_node)
 {
+    if (src_node->get_output_size() != 1)
```

Comment:
If you remove this `if` statement I will approve the PR. This is beyond the scope of the PR. Do a separate PR for this and include Nik as a reviewer as he is the master splicer.